Move some of the FileSystem API inside FileSystem::FileHandle

cdumez opened the pull request:
#### 71722b6d074a9916f2bce4c4d9d567f5daf0a60c
<pre>
Move some of the FileSystem API inside FileSystem::FileHandle
<a href="https://bugs.webkit.org/show_bug.cgi?id=289769">https://bugs.webkit.org/show_bug.cgi?id=289769</a>

Reviewed by Darin Adler.

* Source/WTF/WTF.xcodeproj/project.pbxproj:
* Source/WTF/wtf/FileHandle.cpp:
(WTF::FileSystemImpl::FileHandle::closeIfNecessary):
(WTF::FileSystemImpl::FileHandle::read): Deleted.
(WTF::FileSystemImpl::FileHandle::write): Deleted.
(WTF::FileSystemImpl::FileHandle::lock): Deleted.
(WTF::FileSystemImpl::FileHandle::truncate): Deleted.
(WTF::FileSystemImpl::FileHandle::seek): Deleted.
(WTF::FileSystemImpl::FileHandle::flush): Deleted.
(WTF::FileSystemImpl::FileHandle::id): Deleted.
* Source/WTF/wtf/FileHandle.h:
* Source/WTF/wtf/FileSystem.h:
* Source/WTF/wtf/PlatformGTK.cmake:
* Source/WTF/wtf/PlatformJSCOnly.cmake:
* Source/WTF/wtf/PlatformMac.cmake:
* Source/WTF/wtf/PlatformPlayStation.cmake:
* Source/WTF/wtf/PlatformWPE.cmake:
* Source/WTF/wtf/PlatformWin.cmake:
* Source/WTF/wtf/posix/FileHandlePOSIX.cpp: Added.
(WTF::FileSystemImpl::FileHandle::read):
(WTF::FileSystemImpl::FileHandle::write):
(WTF::FileSystemImpl::FileHandle::truncate):
(WTF::FileSystemImpl::FileHandle::flush):
(WTF::FileSystemImpl::FileHandle::seek):
(WTF::FileSystemImpl::FileHandle::id):
(WTF::FileSystemImpl::FileHandle::close):
(WTF::FileSystemImpl::FileHandle::lock):
(WTF::FileSystemImpl::FileHandle::unlock):
* Source/WTF/wtf/posix/FileSystemPOSIX.cpp:
(WTF::FileSystemImpl::closeFile): Deleted.
(WTF::FileSystemImpl::seekFile): Deleted.
(WTF::FileSystemImpl::truncateFile): Deleted.
(WTF::FileSystemImpl::flushFile): Deleted.
(WTF::FileSystemImpl::writeToFile): Deleted.
(WTF::FileSystemImpl::readFromFile): Deleted.
(WTF::FileSystemImpl::lockFile): Deleted.
(WTF::FileSystemImpl::unlockFile): Deleted.
(WTF::FileSystemImpl::fileID): Deleted.
* Source/WTF/wtf/win/FileHandleWin.cpp: Added.
(WTF::FileSystemImpl::FileHandle::read):
(WTF::FileSystemImpl::FileHandle::write):
(WTF::FileSystemImpl::FileHandle::flush):
(WTF::FileSystemImpl::FileHandle::truncate):
(WTF::FileSystemImpl::FileHandle::seek):
(WTF::FileSystemImpl::FileHandle::id):
(WTF::FileSystemImpl::FileHandle::close):
(WTF::FileSystemImpl::FileHandle::lock):
(WTF::FileSystemImpl::FileHandle::unlock):
* Source/WTF/wtf/win/FileSystemWin.cpp:
(WTF::FileSystemImpl::fileID): Deleted.
(WTF::FileSystemImpl::closeFile): Deleted.
(WTF::FileSystemImpl::seekFile): Deleted.
(WTF::FileSystemImpl::truncateFile): Deleted.
(WTF::FileSystemImpl::flushFile): Deleted.
(WTF::FileSystemImpl::writeToFile): Deleted.
(WTF::FileSystemImpl::readFromFile): Deleted.

Canonical link: <a href="https://commits.webkit.org/292173@main">https://commits.webkit.org/292173@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d134c8db8ab4fb469b78b397e5d3425387f385d9

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/95181 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/14781 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/4639 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/100220 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/45682 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/97229 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/15069 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/23208 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/72596 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/29880 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/98184 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/11256 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/85937 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/52927 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/10965 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/3670 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/45020 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/87852 "Built successfully and passed tests") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/81153 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/3766 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/102262 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/93804 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/22229 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/16214 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/81597 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/22477 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/81953 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/80994 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/20250 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/25557 "Passed tests") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/137/builds/2958 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/15482 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/22199 "Built successfully") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/27325 "Build is in progress. Recent messages:") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/116492 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/21858 "Built successfully") | | [  ~~🧪 jsc-armv7-tests~~](https://ews-build.webkit.org/#/builders/25/builds/33304 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/25331 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/23597 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/WTF/wtf/FileHandle.h:
```diff
@@ -84,13 +85,20 @@ class FileHandle {
     WTF_EXPORT_PRIVATE std::optional<uint64_t> size();
+
+    // Returns the resulting offset from the beginning of the file if successful, -1 otherwise.
     WTF_EXPORT_PRIVATE long long seek(long long offset, FileSeekOrigin);
```
Not great to mix `int64_t` and `long long` in the same class.

darinadler commented on Source/WTF/wtf/posix/FileHandlePOSIX.cpp:
```diff
@@ -0,0 +1,166 @@
+        if (bytesRead >= 0)
+            return bytesRead;
+    } while (errno == EINTR);
+    return -1;
```
Some day we will change this to not use "in band signaling" but I will not push for it now.

darinadler commented on Source/WebCore/contentextensions/SerializedNFA.cpp:
```diff
@@ -49,24 +49,24 @@ bool writeAllToFile(FileSystem::FileHandle& file, const T& container)
 {
-    auto [filename, file] = FileSystem::openTemporaryFile("SerializedNFA"_s);
-    if (!file)
+    auto [filename, fileHandle] = FileSystem::openTemporaryFile("SerializedNFA"_s);
```
Not sure we need to change to the two word variable name. Is `file` too ambiguous?

cdumez replied:
I can go back to `file` if you prefer but keep in mind I renamed all the other variables initialized from `openFile()` to `fileHandle` yesterday. This was to be consistent.

cdumez committed:
Move some of the FileSystem API inside FileSystem::FileHandle
https://bugs.webkit.org/show_bug.cgi?id=289769

Reviewed by Darin Adler.

* Source/WTF/WTF.xcodeproj/project.pbxproj:
* Source/WTF/wtf/FileHandle.cpp:
(WTF::FileSystemImpl::FileHandle::closeIfNecessary):
(WTF::FileSystemImpl::FileHandle::read): Deleted.
(WTF::FileSystemImpl::FileHandle::write): Deleted.
(WTF::FileSystemImpl::FileHandle::lock): Deleted.
(WTF::FileSystemImpl::FileHandle::truncate): Deleted.
(WTF::FileSystemImpl::FileHandle::seek): Deleted.
(WTF::FileSystemImpl::FileHandle::flush): Deleted.
(WTF::FileSystemImpl::FileHandle::id): Deleted.
* Source/WTF/wtf/FileHandle.h:
* Source/WTF/wtf/FileSystem.h:
* Source/WTF/wtf/PlatformGTK.cmake:
* Source/WTF/wtf/PlatformJSCOnly.cmake:
* Source/WTF/wtf/PlatformMac.cmake:
* Source/WTF/wtf/PlatformPlayStation.cmake:
* Source/WTF/wtf/PlatformWPE.cmake:
* Source/WTF/wtf/PlatformWin.cmake:
* Source/WTF/wtf/posix/FileHandlePOSIX.cpp: Added.
(WTF::FileSystemImpl::FileHandle::read):
(WTF::FileSystemImpl::FileHandle::write):
(WTF::FileSystemImpl::FileHandle::truncate):
(WTF::FileSystemImpl::FileHandle::flush):
(WTF::FileSystemImpl::FileHandle::seek):
(WTF::FileSystemImpl::FileHandle::id):
(WTF::FileSystemImpl::FileHandle::close):
(WTF::FileSystemImpl::FileHandle::lock):
(WTF::FileSystemImpl::FileHandle::unlock):
* Source/WTF/wtf/posix/FileSystemPOSIX.cpp:
(WTF::FileSystemImpl::closeFile): Deleted.
(WTF::FileSystemImpl::seekFile): Deleted.
(WTF::FileSystemImpl::truncateFile): Deleted.
(WTF::FileSystemImpl::flushFile): Deleted.
(WTF::FileSystemImpl::writeToFile): Deleted.
(WTF::FileSystemImpl::readFromFile): Deleted.
(WTF::FileSystemImpl::lockFile): Deleted.
(WTF::FileSystemImpl::unlockFile): Deleted.
(WTF::FileSystemImpl::fileID): Deleted.
* Source/WTF/wtf/win/FileHandleWin.cpp: Added.
(WTF::FileSystemImpl::FileHandle::read):
(WTF::FileSystemImpl::FileHandle::write):
(WTF::FileSystemImpl::FileHandle::flush):
(WTF::FileSystemImpl::FileHandle::truncate):
(WTF::FileSystemImpl::FileHandle::seek):
(WTF::FileSystemImpl::FileHandle::id):
(WTF::FileSystemImpl::FileHandle::close):
(WTF::FileSystemImpl::FileHandle::lock):
(WTF::FileSystemImpl::FileHandle::unlock):
* Source/WTF/wtf/win/FileSystemWin.cpp:
(WTF::FileSystemImpl::fileID): Deleted.
(WTF::FileSystemImpl::closeFile): Deleted.
(WTF::FileSystemImpl::seekFile): Deleted.
(WTF::FileSystemImpl::truncateFile): Deleted.
(WTF::FileSystemImpl::flushFile): Deleted.
(WTF::FileSystemImpl::writeToFile): Deleted.
(WTF::FileSystemImpl::readFromFile): Deleted.

Canonical link: https://commits.webkit.org/292173@main
